### PR TITLE
Fixing a bug when splitting FeedRanges into subranges

### PR DESCRIPTION
**Pull request by FabianMeiswinkel**

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/feedranges/FeedRangeEpkImpl.java [hunk 1]
```diff
@@ -39,6 +39,9 @@
 
     public FeedRangeEpkImpl(final Range<String> range) {
         checkNotNull(range, "Argument 'range' must not be null");
+        if (range.getMin().compareTo(range.getMax()) > 1) {
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/feedranges/FeedRangeEpkImpl.java:42]: shouldn't this be `>0` ?

```suggestion
        if (range.getMin().compareTo(range.getMax()) > 0) {
```

**Review by moderakh — APPROVED** [review]
LGTM. Thanks Fabian

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/feedranges/FeedRangeEpkImpl.java (hunk 1 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/feedranges/FeedRangeEpkImpl.java:42]: Yes - interesting - the exception was actually thrown - but i guess that was just luck

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/feedranges/FeedRangeEpkImpl.java [hunk 2]
```diff
@@ -39,6 +39,9 @@
 
     public FeedRangeEpkImpl(final Range<String> range) {
         checkNotNull(range, "Argument 'range' must not be null");
+        if (range.getMin().compareTo(range.getMax()) > 0) {
+            throw new IllegalArgumentException("The provided range is incorrect min is larger than max");
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/feedranges/FeedRangeEpkImpl.java:43]: To be on the safer side, I would suggest using `compareToIgnoreCase` - as I am not sure if backend would always return range in upper case. Just a thought.

**Review by kushagraThapar — COMMENTED** [review]
@FabianMeiswinkel  - please see my comment, not sure if it is worth requesting changes, but wanted to give you a heads up.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/feedranges/FeedRangeEpkImpl.java (hunk 2 above)
Comment (kushagraThapar) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/feedranges/FeedRangeEpkImpl.java:43]: I checked the code again, at other places, we are using `compareTo()`, so I guess we will keep it the same here for consistency, doesn't make sense to introduce any weird behavior.

Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/feedranges/FeedRangeEpkImpl.java:43]: Yes - I actually checked that we will always use upper case intentionally to not have to use any ignore case comparisons. The "string" is always HEX for the range (at least in reality - why the class was made a Generic in the first place when it is only ever being used as Range<String> is a different question - it spans .Net and Java as well)

**Pull request merged**